packages/eslint-config-airbnb/rules/react.js

```diff
@@ -10,7 +10,7 @@ module.exports = {
   'rules': {
     // Prevent missing displayName in a React component definition
     // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/display-name.md
-    'react/display-name': [0, { 'acceptTranspilerName': false }],
+    'react/display-name': [0, { 'ignoreTranspilerName': false }],
```

Comment:
Sadly this setting doesn't appear to differentiate between ES6 classes/stateless functional components - where we *do* want to use the constructor's/function's name rather than having an explicit `displayName` - and `React.createClass`, where we want an explicit `displayName`. 

Given that, and that other rules forbid `React.createClass` anyways, I think `false` is the best setting here.